OWL2/OWL22CASL.hs

```diff
@@ -414,37 +415,37 @@ mapComObjectPropsList cSig mol props a b = do
         Just ol -> fmap (`mkPairs` fs) $ mapObjProp cSig ol a b
 
 -- | mapping of Data Range
-mapDataRange :: CASLSign -> DataRange -> Int -> Result (CASLFORMULA, CASLSign)
+mapDataRange :: CASLSign -> DataRange -> Int -> Result (CASLFORMULA, [CASLSign])
 mapDataRange cSig dr i = case dr of
     DataType d fl -> do
         let dt = mkMember (qualData i) $ uriToCaslId d
         (sens, s) <- mapAndUnzipM (mapFacet cSig i) fl
-        return (conjunct $ dt : sens, uniteL $ cSig : s)
+        return (conjunct $ dt : sens, concat s)
     DataComplementOf drc -> do
         (sens, s) <- mapDataRange cSig drc i
-        return (mkNeg sens, uniteCASLSign cSig s)
+        return (mkNeg sens, s)
     DataJunction jt drl -> do
         (jl, sl) <- mapAndUnzipM ((\ s v r -> mapDataRange s r v) cSig i) drl
-        let usig = uniteL sl
+        --let usig = uniteL sl
         return $ case jt of
-                IntersectionOf -> (conjunct jl, usig)
-                UnionOf -> (disjunct jl, usig)
+                IntersectionOf -> (conjunct jl, concat sl)
+                UnionOf -> (disjunct jl, concat sl)
     DataOneOf cs -> do
         ls <- mapM mapLiteral cs
-        return (disjunct $ map (mkStEq $ qualData i) ls, cSig)
+        return (disjunct $ map (mkStEq $ qualData i) ls, [])
 
 mkFacetPred :: TERM f -> ConstrainingFacet -> Int -> (FORMULA f, Id)
 mkFacetPred lit f var =
     let cf = mkInfix $ fromCF f
     in (mkPred dataPred [qualData var, lit] cf, cf)
 
 mapFacet :: CASLSign -> Int -> (ConstrainingFacet, RestrictionValue)
-    -> Result (CASLFORMULA, CASLSign)
+    -> Result (CASLFORMULA, [CASLSign])
 mapFacet sig var (f, r) = do
     con <- mapLiteral r
     let (fp, cf) = mkFacetPred con f var
-    return (fp, uniteCASLSign sig $ (emptySign ())
-            {predMap = MapSet.fromList [(cf, [dataPred])]})
+    return (fp, [(emptySign ())
+            {predMap = MapSet.fromList [(cf, [dataPred])]}])
 
```

Comment:
shouldn't `sig` appear in the resulting signature list?
why s the empty sig needed in the list?

Reply:
no, `sig` should actually be omitted from all these functions. I will do this. 
I kept the old formatting and it's hard to read, it should be the signature that you get by inserting (cf, dataPred) in the predMap of the empty signature.